packages/cow-hooks-ui/src/hooks/useSubmitHook.ts

```diff
@@ -10,18 +10,21 @@ export function useSubmitHook({
   context,
   publicClient,
   recipientOverride,
+  defaultGasLimit,
 }: {
   actions: CoWHookDappActions | undefined;
   context: HookDappContextAdjusted | undefined;
   publicClient: PublicClient | undefined;
   recipientOverride?: string;
+  defaultGasLimit?: bigint;
 }) {
   return useCallback(
     async (hook: Omit<CowHook, "gasLimit">) => {
-      if (!context || !actions) return;
+      if (!context || !actions || !publicClient)
+        throw new Error("Missing context");
 
       const estimatedGas = await publicClient
-        ?.estimateGas({
```

Comment:
think we wont need this if a default gasLimit is passed, just remember adding this "if" here

Reply:
Yes, just wants to confirm it with them before did it.